src/Naming.js

```diff
@@ -0,0 +1,16 @@
+import renderTag from './renderTag';
+import ClassNameBuilder from './ClassNameBuilder';
+
+export default {
+    __constructor() {
+        this.__base(...arguments);
+        this.__cnb = new ClassNameBuilder(this._naming);
+        this.__render = renderTag(this.__cnb);
+    },
+
+    _naming : {
```

Comment:
maybe we need to use more difficult field name, ex: `__dengerouslyNamingSettings`